app/harvesters/cdl_harvester.rb

```diff
@@ -0,0 +1,50 @@
+##
+# A harvester for CDL's Solr API-endpoint
+#
+#
+# @see Krikri::Harvesters::ApiHarvester
+class CdlHarvester < Krikri::Harvesters::ApiHarvester
+ include Krikri::Harvester
+  attr_reader :opts
+
+  ##
+  # @param opts [Hash] options for the harvester
+  # @see .expected_opts
+  def initialize(opts = {})
+    opts[:uri] ||= 'https://solr.calisphere.org/solr/query'
+    opts[:name] ||= 'cdl'
+    super
+    @opts['params'] ||= { 'q' => '*:*' }
+  end
+
+  ##
+  # @return [Hash] A hash documenting the allowable options to pass to
+  #   initializers.
+  #
+  # Http headers are passed in at the same level as params inside
+  # the opts hash.
+  #
+  # @see Krikri::Harvester::expected_opts
+  def self.expected_opts
+    {
+      key: :api,
+        opts: {
+          authentication: { type: :string, required: true },
+          params: { type: :string, required: false }
+       }
+    }
+  end
+
+  private
+
+  ##
+  # Overrides ApiHarvester.get_identifier because the idenitifer field
+  # for CDL is different from MDL's.
+  #
+  # @param doc [#to_s] a raw record document with an identifier
```

Comment:
@no-reply or is `#to_s` valid because `#get_content` is calling `#to_json` on the hash object which creates a string serialization of the json?